sophiread/SophireadCLI/src/sophiread.cpp

```diff
@@ -1,137 +1,120 @@
+    spdlog::info("  -E <output_events>       Output events HDF5 file");
+    spdlog::info("  -u <config_file>         User configuration JSON file (optional)");
+    spdlog::info("  -T <tof_imaging_folder>  Output folder for TIFF TOF images (optional)");
+    spdlog::info("  -f <tof_filename_base>   Base name for TIFF files (default: tof_image)");
+    spdlog::info("  -m <tof_mode>            TOF mode: 'hit' or 'neutron' (default: neutron)");
+    spdlog::info("  -d                       Enable debug logging");
+    spdlog::info("  -v                       Enable verbose logging");
 }
 
-/**
- * @brief Timed find TPX3H.
- *
- * @param[in] rawdata
- * @return std::vector<TPX3>
- */
-std::vector<TPX3> timedFindTPX3H(const std::vector<char> &rawdata) {
-  auto start = std::chrono::high_resolution_clock::now();
-  auto batches = findTPX3H(rawdata);
-  auto end = std::chrono::high_resolution_clock::now();
-  auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
-  spdlog::info("Locate all headers: {} s", elapsed / 1e6);
-
-  return batches;
-}
-
-/**
- * @brief Timed locate timestamp.
- *
- * @param[in, out] batches
- * @param[in] rawdata
- */
-void timedLocateTimeStamp(std::vector<TPX3> &batches, const std::vector<char> &rawdata) {
-  auto start = std::chrono::high_resolution_clock::now();
-  unsigned long tdc_timestamp = 0;
-  unsigned long long gdc_timestamp = 0;
-  unsigned long timer_lsb32 = 0;
-  for (auto &tpx3 : batches) {
-    updateTimestamp(tpx3, rawdata, tdc_timestamp, gdc_timestamp, timer_lsb32);
-  }
-  auto end = std::chrono::high_resolution_clock::now();
-  auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
-  spdlog::info("Locate all timestamps: {} s", elapsed / 1e6);
-}
+ProgramOptions parse_arguments(int argc, char* argv[]) {
+    ProgramOptions options;
+    int opt;
+
+    while ((opt = getopt(argc, argv, "i:H:E:u:T:f:m:dv")) != -1) {
+        switch (opt) {
+            case 'i':
+                options.input_tpx3 = optarg;
+                break;
+            case 'H':
+                options.output_hits = optarg;
+                break;
+            case 'E':
+                options.output_events = optarg;
+                break;
+            case 'u':
+                options.config_file = optarg;
+                break;
+            case 'T':
+                options.output_tof_imaging = optarg;
+                break;
+            case 'f':
+                options.tof_filename_base = optarg;
+                break;
+            case 'm':
+                options.tof_mode = optarg;
+                break;
+            case 'd':
+                options.debug_logging = true;
+                break;
+            case 'v':
+                options.verbose = true;
+                break;
+            default:
+                print_usage(argv[0]);
+                throw std::runtime_error("Invalid argument");
```

Comment:
Usability (not urgent): add the invalid argument supplied to the error

Reply:
updated to make it easier for user to use